server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java

```diff
@@ -217,11 +222,30 @@ private void confirmDeletes(GarbageCollectionEnvironment gce,
         throw new RuntimeException(
             "Scanner over metadata table returned unexpected column : " + entry.getKey());
     }
+    Set<String> tableIdsAfter = gce.getTableIDs();
+    ensureAllTablesChecked(tableIdsBefore, tableIdsSeen, tableIdsAfter);
 
     confirmDeletesFromReplication(gce.getReplicationNeededIterator(),
         candidateMap.entrySet().iterator());
   }
 
+  @VisibleForTesting
+  protected void ensureAllTablesChecked(Set<String> tableIdsBefore, Set<String> tableIdsSeen,
+      Set<String> tableIdsAfter) {
+
+    // if a table was added or deleted during this run, it is acceptable to not
+    // have seen those tables ids when scanning the metadata table. So get the intersection
+    Set<String> tableIdsMustHaveSeen = new HashSet<>(tableIdsBefore);
+    tableIdsMustHaveSeen.retainAll(tableIdsAfter);
```

Comment:
The following is a suggestion based on discussion we had at the top level of the PR.  This may be another useful sanity check, but I have not fully convinced myself it will not cause troublesome false positives. 

```suggestion
    tableIdsMustHaveSeen.retainAll(tableIdsAfter);
    
        if(tableIdsMustHaveSeen.isEmpty() && !tableIdsSeen.isEmpty()) {
    	// we saw no table ids in ZK but did in the metadata table.  This is unexpected.
    	throw new RuntimeException("Saw no table ids in ZK but did see table ids in metadata table: " + tableIdsSeen);
    }
```

Reply:
Will apply this change, thanks

Reply:
`retainAll` returns a boolean to signify that the collection changed. You could add it to the if statement.

```suggestion
    if (tableIdsMustHaveSeen.retainAll(tableIdsAfter) && tableIdsMustHaveSeen.isEmpty() && !tableIdsSeen.isEmpty()) {
```

Reply:
When `tableIdsMustHaveSeen` is empty it seems like `retainAll()` would always return false. We want to check that it is empty, so adding it to the if may mean `tableIdsMustHaveSeen.retainAll(tableIdsAfter) && tableIdsMustHaveSeen.isEmpty()` can never be true.